crates/proof-of-sql/src/sql/proof/query_result.rs

```diff
@@ -39,6 +39,30 @@ pub enum QueryError {
     InvalidColumnCount,
 }
 
+impl From<TableCoercionError> for QueryError {
+    fn from(error: TableCoercionError) -> Self {
+        match error {
+            TableCoercionError::ColumnCoercionError {
+                source: ColumnCoercionError::Overflow,
+            } => QueryError::Overflow,
+            TableCoercionError::ColumnCoercionError {
+                source: ColumnCoercionError::InvalidTypeCoercion,
+            } => ProofError::VerificationError {
+                error: "Invalid Type Coercion",
+            }
+            .into(),
+            TableCoercionError::NameMismatch => ProofError::VerificationError {
+                error: "Field Names Mismatch",
+            }
+            .into(),
+            TableCoercionError::ColumnCountMismatch => ProofError::VerificationError {
```

Comment:
This is evidence that we really should fix `ProofError` by making `TableCoercionError` an enum variant

Reply:
In this case `TableCoercionError` shouldn't be a direct variant since some of the errors are verification errors, and some are not. We want a hard distinction between the two. So, even if add variants to `ProofError`, we still need a custom convert here.